docs/design/resources.md

```diff
@@ -0,0 +1,120 @@
+# Operator Resources
+
+The etcd cluster operator extends the Kubernetes API with two new
+resource definitions:
+
+* `EtcdCluster`, or the "cluster resource", defines an entire
+  operating cluster.
+* `EtcdPeer`, or the "peer resource", defines a single peer of the
+  etcd cluster. This principally concerns a Pod, but also includes
+  attached storage and network identity.
+
+## The Peer Resource
+
+The peer resource represents a single peer in an etcd
+cluster. Encoding its bootstrap instructions, persistence settings,
+and extra pod settings (e.g., overridden CPU limits). Its `status`
+field exposes information about the peer including its hostname. In
+future the status may also include the liveness status if it's
+currently the leader of the etcd cluster.
+
+When reconciling the etcd peer resource the operator will do the
```

Comment:
does "the operator" here mean
a) a singleton multi-tenant operator for all EtcdClusters (and therefore all EtcdPeers) in the k8s cluster
b) an individual operator for the specific EtcdCluster
c) an individual operator for the specific EtcdPeer

Reply:
> a) a singleton multi-tenant operator for all EtcdClusters (and therefore all EtcdPeers) in the k8s cluster

This one 😄 I think the isolation model should be per namespace if we want to use that (and kubebuilder gives us the option to isolate the operator to a single namespace with the `--namespace` flag).

Reply:
makes sense. confused me a little as "The peer operator" is mentioned further down this doc, suggesting that peers would have their own operators.

Reply:
Sorry, I was unclear here. My intent is that we'll ship a single `etcd-cluster-operator` binary (and `improbable/etcd-cluster-operator` Docker image) that will run both the `EtcdCluster` and `EtcdPeer` reconciliation. I also expected that we'd run a singleton multi-tenant deployment of the operator to handle all namespaces in the cluster. (I also kind of expected we'd deploy the operator in a different namespace to the cluster it manages.)

Would it help if I clarified the language used? Or perhaps added a section explaining deployment and tenancy?

Reply:
I added a draft deployment section in 93c2df5 which hopefully clarifies a bit.

Reply:
looks good, thanks